File: 2 Revalida en C++/5 Funciones/2.cpp
///funcion para saber si un numero es perfecto
#include <bits/stdc++.h>
using namespace std;
int es_perfecto(int n)
{
    int suma=0;
	for (int i=1;i<=n/2;i++)
    {
        if ((n%i)==0)
            suma=suma+i;
    }
	if (suma==n)
		return true;
	else
		return false;
}
int main()
{
    for(int n=1;n<=100;n++)
    {
        if(es_perfecto(n))
        {
            cout<<"los numeros perfectos son="<<n<<endl;
        }
    }
    return 0;
}

File: 1 Revalida en python/5 Funciones/2.py
 #funcion para saber si un numero es perfecto
def es_perfecto(n):
	suma=0
	i=1
	for i in (1,(n/2)):
		if ((n%i)==0):
			suma=suma+1
	if (suma==n):
		return True
	else:
		return False
#numeros perfectos entre el 1 al 100
n=1
for n in [100]:
	if(es_perfecto(n)):
		print (n)